### DISCUSSION TITLE
Is it possible to control the start time of workflow?

### DISCUSSION THREAD

**Discussion by ghost** [nextflow-io/nextflow#4268] [Q&A] [answered]
Hi,

I want to specify the start time for pipeline, but I don't know how to implement.
I tried the following, but in the end it was recognized as one workflow, so I couldn't specify the start time properly.
```
process wait_until_time {
    input:
    val starttime
    output:
    val "the starttime", emit:wait_is_over
    script:
    """
    current_time=\$(date "+%Y%m%d%H%M")

    if [ "${starttime.substring(1)}" -lt "\$current_time" ]; then
    exit 0
    fi

    while [ "\$(date "+%Y%m%d%H%M")" -lt "${starttime}" ]; do
    sleep 1
    done
    exit 0
    """
}
```
```
workflow MAIN_WORKFLOW {
 take: wait_over
 main:
 # main processes
}
```
```
workflow {
  wait_until_time(params.time)
  MAIN_WORKFLOW(wait_until_time.out.wait_is_over)
}
```
I understand that this kind of implementation goes against the philosophy of Nextflow, which has a strong focus on data flow, but is there any way to do this?

Thanks in advance.

**Comment by bentsherman** [comment]
Checkout `workflow.start`: https://nextflow.io/docs/edge/metadata.html#runtime-metadata

**Reply by ghost** [reply]
I thought this is an object that can store the start time, not an option to specify the start time.
Conversely, if I specify this in nextflow.config, does this mean that the start time will be set based on this?

**Reply by bentsherman** [reply]
Well if you want to *set* the workflow start time then you have to... start the workflow at the desired time 😄 

Basically you need to use an external tool or service to schedule the workflow run, like cron as Marcel suggested.

**Comment by mribeirodantas** [comment, marked as answer]
I think your best option is to use something like [cron](https://www.freecodecamp.org/news/cron-jobs-in-linux/) for that.

**Comment by ghost** [comment]
I will specify it with the `at` command. Thank you for comments and pointing out.
